lib/SimpleSAML/Memcache.php

```diff
@@ -304,24 +306,27 @@ private static function addMemcacheServer($memcache, $server)
      *
      * @param array $group Array of servers which should be created as a group.
      *
-     * @return \Memcache A Memcache object of the servers in the group
+     * @return \Memcache|\Memcached A Memcache object of the servers in the group
      *
      * @throws \Exception If the servers configuration is invalid.
      */
     private static function loadMemcacheServerGroup(array $group)
     {
-        $class = class_exists('\Memcache') ? '\Memcache' : (class_exists('\Memcached') ? '\Memcached' : false);
-        if (!$class) {
+        if (class_exists(_Memcache::class)) {
+            $memcache = new _Memcache();
+            self::$extension = strtolower(_Memcache::class);
+        } elseif (class_exists(_Memcached::class)) {
+            $memcache = new _Memcached();
+            self::$extension = strtolower(_Memcached::class);
+        } else {
             throw new \Exception(
                 'Missing Memcached implementation. You must install either the Memcache or Memcached extension.'
             );
-        } elseif (strtolower($class) === '\memcache') {
-            Logger::warning("The use of PHP-extension memcache is deprecated. Please migrate to the memcached extension.");
         }
-        self::$extension = strtolower($class);
 
-        // create the \Memcache object
-        $memcache = new $class();
+        if (self::$extension === '\memcache') {
```

Comment:
Is there a reason for using lowercase here? Can't we just compare with `\Memcache`?